soc/nxp/mcx/mcxc/Kconfig

```diff
@@ -0,0 +1,148 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_MCXC
+	select ARM
+	select CPU_CORTEX_M0PLUS
+	select CPU_CORTEX_M_HAS_SYSTICK
+	select CPU_CORTEX_M_HAS_VTOR
+	select CLOCK_CONTROL
+	select PLATFORM_SPECIFIC_INIT
+	select HAS_MCUX
+	select HAS_MCUX_ADC16
+	select HAS_MCUX_SIM
+	select HAS_OSC
+	select HAS_MCG
+	select HAS_MCUX_RCM
+	select HAS_MCUX_FTFX
+	select HAS_MCUX_LPUART
+	select HAS_MCUX_LPI2C
+	select HAS_MCUX_TPM
+	
+if SOC_SERIES_MCXC
+
+config HAS_OSC
+	bool
+	help
+	  Set if the oscillator (OSC) module is present in the SoC.
+
+if HAS_OSC
+
+choice MCXC_OSC_MODE
+	prompt "Oscillator Mode Selection"
+	default MCXC_OSC_EXTERNAL
+
+config MCXC_OSC_EXTERNAL
+	bool "External reference clock"
+	help
+	  Set this option to use the oscillator in external reference clock mode.
+
+config MCXC_OSC_LOW_POWER
+	bool "Low power oscillator"
+	help
+	  Set this option to use the oscillator in low-power mode.
+
+config MCXC_OSC_HIGH_GAIN
+	bool "High gain oscillator"
+	help
+	  Set this option to use the oscillator in high-gain mode.
```

Comment:
move to dts

Reply:
fix this in the original commit not adding something wrong then changing to to be right in a later commit